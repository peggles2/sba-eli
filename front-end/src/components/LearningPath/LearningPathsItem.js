import React, { Component } from 'react'
import { Card, Image } from "semantic-ui-react";
import LearningPathProgress from "./LearningPathProgress";
import axios from "axios";
import { withRouter } from 'react-router-dom';
import { connect } from "react-redux";

import { getLearningEvents } from "../../actions/learningEventActions";

export class LearningPathsItem extends Component {
  state = { topicsList: [] }

  componentDidMount() {
    this.getTopicsList();
  }

  learningPathProgress = (total) => {
    // const isUserLoggedIn = this.props.isUserLoggedIn
    const isUserLoggedIn = true;
    if(isUserLoggedIn) {
      const complete = 0;

      return (
        <Card.Content>
          <LearningPathProgress complete={complete} total={total}/>
        </Card.Content>
      );
    }
  };

  getTopicsList = () => {
    const baseUrl = process.env.REACT_APP_SERVICE_HOST;

    axios.get(baseUrl + `/learning_objectives/`, {params: {course_id: this.props.id}})
      .then(res => {
        const topicsList = res.data
        this.setState({ topicsList })
      })
  }

  totalMicroLearningEvents = () => {
    const reducer = (totalList, number) => totalList + number;
    const topicsList = this.state.topicsList;

    if(topicsList.length > 0) {
      const total = topicsList.map(topic => topic.items_count).reduce(reducer);

      return total;
    } else {
      return 0
    }
  }

  render() {
    const total = this.totalMicroLearningEvents();
    const learningPathDescription = "Maybe it means something more - something we can't yet understand."

    return(
      <Card>
        <Image
          src={`/Image_Placeholder.png`}
          alt="learning path image placeholder"
        />
        <Card.Content>
          <Card.Header aria-label={`Link to Learning Path ` + this.props.name}>
            <a href="#" onClick={() => { this.props.history.push('/learning_paths/' + this.props.id)}}>{this.props.name}</a>
          </Card.Header>
          <Card.Meta>{this.props.course_code}</Card.Meta>
          <Card.Description>{learningPathDescription}</Card.Description>
        </Card.Content>
        {this.learningPathProgress(total)}
      </Card>
    )
  }
};

const mapStateToProps = (store) => {
  return {
    isUserLoggedIn: store.login.isUserLoggedIn,
  }
}

export default withRouter(connect(mapStateToProps)(LearningPathsItem));
