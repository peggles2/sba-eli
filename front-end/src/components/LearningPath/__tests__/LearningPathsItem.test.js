import React from "react";

import { shallow, mount } from "enzyme";
import { LearningPathsItem } from '../LearningPathsItem'

describe("LearningPathsItem", () => {
  it("should render a Card", () => {
    const wrapper = shallow(<LearningPathsItem learningPaths={[]}/>);
    expect(wrapper.find("Card").length).toBe(1);
  });

  it("has a placeholder", () => {
    const wrapper = shallow(<LearningPathsItem learningPaths={[]}/>);
    expect(wrapper.find("Image").length).toBe(1);
  });

  it("should have a link to the learning path", () => {
    const history = [];
    const wrapper = mount(<LearningPathsItem learningPaths={[]} history={history} id={1}/>);
    
    let href = wrapper.find('a').first().getDOMNode().getAttribute('href');
    expect(href).toBe('/learning_paths/1');
  })  
});