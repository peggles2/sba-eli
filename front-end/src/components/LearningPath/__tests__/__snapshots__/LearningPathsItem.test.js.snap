// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearningPathsItem should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LearningPathsItem
    item={
      Object {
        "id": 1,
        "name": "LP 1",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <a
        href="/learning_paths/undefined"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "href": "/learning_paths/undefined",
      },
      "ref": null,
      "rendered": null,
      "type": "a",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <a
          href="/learning_paths/undefined"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "href": "/learning_paths/undefined",
        },
        "ref": null,
        "rendered": null,
        "type": "a",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
